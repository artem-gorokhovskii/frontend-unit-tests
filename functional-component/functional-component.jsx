import React from 'react';

export const Counter = ({ name }) => {
    const [pressedTimes, setPressedTimes] = React.useState(0);
    return (
        <div className="counter">
            <div className='greetings'>Привет, {name}!</div>
            <div className='message'>Вы нажали {pressedTimes} раз</div>
            {pressedTimes >= 3 && <div className='notification'>Кликов больше 2!</div>}
            <button onClick={() => setPressedTimes(prevTimes => prevTimes + 1)}></button>
        </div>
    );
};


