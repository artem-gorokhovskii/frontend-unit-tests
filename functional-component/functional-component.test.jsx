import React from 'react';

import { shallow } from 'enzyme';
import { Counter } from './functional-component';

const MIN_PROPS = { name: 'Artem Gorokhovskii' };

describe('Functional component Counter', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Counter {...MIN_PROPS} />);
    });
    it('Должен иметь корректный снапшот', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('Приветствие выводится корректно', () => {
        const text = wrapper.find('.greetings').text();
        expect(text).toBe(`Привет, ${MIN_PROPS.name}!`);
    });
    it('Нажатие кнопки увеличит счетчик и выведет корректное сообщение', () => {
        const prevMessage = wrapper.find('.message').text();

        const button = wrapper.find('button');
        button.simulate('click');
        
        const message = wrapper.find('.message').text();

        expect(prevMessage).toBe('Вы нажали 0 раз');
        expect(message).toBe('Вы нажали 1 раз');
    });
    it('Нажатие кнопки 3 раза выведет предупреждение', () => {
        const prevNotificationExists = !!wrapper.find('.notification').length;

        const button = wrapper.find('button');
        button.simulate('click');
        button.simulate('click');
        button.simulate('click');
        
        const notificationExists = wrapper.find('.notification').length;
        
        expect(prevNotificationExists).toBe(false);
        expect(notificationExists).toBe(1);
    });
});