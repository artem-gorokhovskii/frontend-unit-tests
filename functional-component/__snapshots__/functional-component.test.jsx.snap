// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional component Counter Должен иметь корректный снапшот 1`] = `
"<div className=\\"counter\\">
  <div className=\\"greetings\\">
    Привет, 
    Artem Gorokhovskii
    !
  </div>
  <div className=\\"message\\">
    Вы нажали 
    0
     раз
  </div>
  <button onClick={[Function: onClick]} />
</div>"
`;
